forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,227 @@
+
+    def _cache_key_function(self, line_info: Tuple[int, List[str]]) -> str:
+        return str(line_info[0])
+
+    def _parse_pack(
+        self, line_info: Tuple[int, List[str]]
+    ) -> Iterator[DataPack]:
+        line_id, line = line_info
+        # content label title
+        pack = DataPack()
+
+        # subtext fields must follow the ontology names
+        data_fields = self.configs.data_fields
+
+        if len(self.configs.data_fields) != len(line):
+            raise ProcessorConfigError(
+                "Data fields provided in config "
+                "is not aligned with the actual line info from dataset.\n"
+                "Data fields length: "
+                + str(len(self.configs.data_fields))
+                + "\n",
+                "Line length: " + str(len(line)),
+            )
+
+        df_dict = OrderedDict()
+        for df, value in zip(data_fields, line):
+            df_dict[
+                df
+            ] = value  # in general, value can be subtext or other types of data
+
+        # it determines the order of concatenation
+        subtext_fields = self.configs.subtext_fields
+
+        # get text and subtext indices
+        text, subtext_indices = generate_text_n_subtext_indices(
+            subtext_fields, df_dict
+        )
+        self.set_text(pack, text)
+        if df_dict["label"].isdigit() != self.configs.digit_label:
+            raise ProcessorConfigError(
+                "Label format from dataset "
+                "is not consistent with the label format from configs. \n"
+                + "dataset digit label status: "
+                + str(df_dict["label"].isdigit())
+                + "\n"
+                "config digit label status: " + str(self.configs.digit_label)
+            )
+
+        if self.configs.digit_label:
+            if self.configs.one_based_index_label:
+                # adjust label encoding if dataset digit label is one-based
+                class_id = int(df_dict["label"]) - 1
+            else:
+                class_id = int(df_dict["label"])
+        else:
+            # decode string label from dataset into digit label
+            class_id = self._class2index[df_dict["label"]]
+        # initialize all subtexts instances as data pack module
+        # to add corresponding variables and functions
+        for subtext_field, (start_idx, end_idx) in subtext_indices.items():
+            path_str, module_str = subtext_field.rsplit(".", 1)
+            mod = importlib.import_module(path_str)  # sentence ontology module
+            entry_class = getattr(mod, module_str)
+            entry_class(pack, start_idx, end_idx)
+        # for now, we use document to store concatenated text and set the class here
+        doc = Document(pack, 0, subtext_indices[subtext_fields[-1]][1])
+        doc.document_class = [self.configs.index2class[class_id]]
+
+        pack.pack_name = line_id
+        yield pack
+
+    @classmethod
+    def default_configs(cls):
```

Comment:
Maybe it's better to reorganize the inline comments below into a doctoring. You can refer to https://github.com/asyml/forte/blob/9597db37e21fb998f7e235bbbd66b93f1924e64b/forte/processors/base/writers.py#L126